### PR TITLE
feat: Tokenizer support build with params and clone for concurrency

### PR DESCRIPTION
**Pull request by aoiasd**
relate: https://github.com/milvus-io/milvus/issues/35853 https://github.com/milvus-io/milvus/issues/36751

### CODE REVIEW

File: internal/core/thirdparty/tantivy/tantivy-binding/src/index_reader_text_c.rs [hunk 1]
```diff
@@ -18,23 +24,24 @@ pub extern "C" fn tantivy_match_query(ptr: *mut c_void, query: *const c_char) ->
...
             (*real).register_tokenizer(
                 String::from(tokenizer_name_str.to_str().unwrap()),
                 text_analyzer,
             );
         },
-        None => {
-            panic!("unsupported tokenizer");
-        }
+        Err(err) => {
+            panic!("create tokenizer failed with error: {}", err.to_string());
```
Comment (zhengbuqian) [inline internal/core/thirdparty/tantivy/tantivy-binding/src/index_reader_text_c.rs:44]: does golang catch this panic and return to user?

File: internal/core/thirdparty/tantivy/tantivy-binding/src/tokenizer.rs [hunk 2]
```diff
@@ -1,54 +1,201 @@
...
-    static ref DEFAULT_TOKENIZER_MANAGER: TokenizerManager = TokenizerManager::default();
+use crate::tokenizer_filter::*;
+use crate::error::TantivyError;
+
+pub(crate) fn standard_analyzer() -> TextAnalyzer {
+    standard_builder().build()
+}
+
+fn standard_builder() -> TextAnalyzerBuilder{
+    TextAnalyzer::builder(SimpleTokenizer::default()).dynamic()
```
Comment (zhengbuqian) [inline internal/core/thirdparty/tantivy/tantivy-binding/src/tokenizer.rs:14]: Seems `SimpleTokenizer` does not use lowercase, let's change it

File: internal/core/thirdparty/tantivy/tantivy-binding/src/tokenizer.rs [hunk 3]
```diff
@@ -1,54 +1,201 @@
...
+impl AnalyzerBuilder<'_>{
+    fn new(params: &json::Map<String, json::Value>) -> AnalyzerBuilder{
+        AnalyzerBuilder{
+            filters: HashMap::new(),
+            params:params,
+        }
+    }
+
+    fn get_tokenizer_name(&self) -> Result<String, TantivyError>{
+        let tokenizer=self.params.get("tokenizer");
```
Comment (zhengbuqian) [inline internal/core/thirdparty/tantivy/tantivy-binding/src/tokenizer.rs:47]:
```suggestion
        let tokenizer=self.params.get("tokenizer");
        if tokenizer.is_none() {
            Ok("standard".to_string())
        }
        if !tokenizer.unwrap().is_string() {
            ...
        }
        Ok(tokenizer.unwrap()...)
```

File: internal/core/thirdparty/tantivy/tantivy-binding/src/tokenizer.rs [hunk 4]
```diff
@@ -1,54 +1,201 @@
...
+        }
+
+        if !tokenizer.is_none(){
+            Ok(tokenizer.unwrap().as_str().unwrap().to_string())
+        }else{
+            Ok("standard".to_string())
+        }
+    }
+
+    fn add_costom_filter(&mut self, name: &String, params: &json::Map<String, json::Value>) -> Result<(),TantivyError>{
```
Comment (zhengbuqian) [inline internal/core/thirdparty/tantivy/tantivy-binding/src/tokenizer.rs:59]: typo custom and several other places

**Review by zhengbuqian — COMMENTED** [review]
this is awesome, thanks!

File: internal/core/thirdparty/tantivy/tantivy-binding/src/index_reader_text_c.rs (hunk 1 above)
Comment (aoiasd) [reply internal/core/thirdparty/tantivy/tantivy-binding/src/index_reader_text_c.rs:44]: No, now all rust error don't return to user, we need a rust status return to c like cstatus return to go, to fetch error.

**Comment by aoiasd** [comment]
rerun cpp-unit-test

**Comment by aoiasd** [comment]
rerun go-sdk

**Comment by aoiasd** [comment]
rerun cpp-unit-test

File: internal/core/thirdparty/tantivy/tantivy-binding/src/tokenizer_filter.rs [hunk 5]
```diff
@@ -46,18 +47,10 @@ fn get_length_filter(params: &json::Map<String, json::Value>) -> Result<SystemFi
 
 fn get_stop_filter(params: &json::Map<String, json::Value>)-> Result<SystemFilter, TantivyError>{
```
Comment (zhengbuqian) [inline internal/core/thirdparty/tantivy/tantivy-binding/src/tokenizer_filter.rs:48]: use full name "stop word filter", same as the error message below

File: internal/core/thirdparty/tantivy/tantivy-binding/src/util.rs [hunk 6]
```diff
@@ -28,3 +30,19 @@ pub fn free_binding<T>(ptr: *mut c_void) {
         drop(Box::from_raw(real));
     }
 }
+
+pub(crate) fn get_stop_words_list(value: &json::Value) -> Result<Vec<String>, TantivyError>{
```
Comment (zhengbuqian) [inline internal/core/thirdparty/tantivy/tantivy-binding/src/util.rs:34]: simply `get_string_list` or even generic `get_json_value_list`. returned error should also be generic: `json::Value is not a list of <type>`, and the caller to add info about stop word to the error msg.

**Review by zhengbuqian — COMMENTED** [review]
thanks a lot!

**Comment by aoiasd** [comment]
rerun go-sdk

**Comment by aoiasd** [comment]
rerun cpp-unit-test

**Comment by aoiasd** [comment]
rerun go-sdk

**Comment by aoiasd** [comment]
rerun cpp-unit-test

**Comment by aoiasd** [comment]
rerun cpp-unit-test

**Comment by aoiasd** [comment]
rerun cpp-unit-test

**Comment by aoiasd** [comment]
rerun go-sdk

**Comment by aoiasd** [comment]
rerun cpp-unit-test

**Comment by aoiasd** [comment]
rerun cpp-unit-test

**Comment by aoiasd** [comment]
rerun go-sdk

**Comment by aoiasd** [comment]
rerun go-sdk

**Pull request merged**